### PR TITLE
test:Add Integration tests for account delete functions

### LINKED ISSUE #6107: test: Write tests for user delete helper functions
**Description**
This is a sub issue of #5320 

Integration Tests are required for user delete helper functions to ensure that they work properly. This increases the test coverage for the code.

**Additional context**
Working on it.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #6107 
Part of #5320 

#### Short description of what this resolves:
 Increases test coverage by adding integrations tests for user delete helper functions

#### Changes proposed in this pull request:

- Added integration tests for user delete helper functions

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
@iamareebjamal @uds5501 @shreyanshdwivedi Please review!

File: tests/all/integration/api/helpers/test_user.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            user = create_user(email="test_user@gmail.com.deleted", password="testpass")
+            save_to_db(user)
+            modified_user = modify_email_for_user_to_be_restored(user)
+            self.assertEqual("test_user@gmail.com", modified_user.email)
+
+            user1 = create_user(email="test_user@gmail.com", password="testpass")
+            save_to_db(user1)
+            user2 = create_user(email="test_user@gmail.com.deleted", password="testpass")
+            save_to_db(user2)
+            self.assertRaises(ForbiddenException, modify_email_for_user_to_be_restored, user2)
```
Comment (shreyanshdwivedi) [inline tests/all/integration/api/helpers/test_user.py]: The tests LGTM 👍
@mrsaicharan1 just a query - Is the user2 passed to the function `modify_email_for_user_to_be_restored` by this line?

Comment (mrsaicharan1) [reply tests/all/integration/api/helpers/test_user.py]: @shreyanshdwivedi yes, that's how `assertRaises` works!

**Comment by mrsaicharan1** [comment]
@iamareebjamal @CosmicCoder96 Please review.

File: tests/all/integration/api/helpers/test_user.py (hunk 1 above)
Comment (shreyanshdwivedi) [reply tests/all/integration/api/helpers/test_user.py]: Cool. Approving

**Comment by mrsaicharan1** [comment]
@niranjan94 @CosmicCoder96 PLease have a look

**Pull request merged**